up_client_socket/cpp/conanfile.py

```diff
@@ -0,0 +1,64 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir
+import os
+
+class UpClientSocket(ConanFile):
+    name = "up_client_socket"
+    package_type = "library"
+    license = "Apache-2.0 license"
+    homepage = "https://github.com/eclipse-uprotocol"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "C++ up client socket library for testagent"
+    #topics = ("ulink client", "transport")
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    conan_version = None
+    generators = "CMakeDeps"
+    version = "0.1"
+    exports_sources = "CMakeLists.txt", "src/*" 
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_cross_compiling": [True, False],
+        "build_testing": [True, False],
+        "build_unbundled": [True, False],
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": False,
+        "build_cross_compiling": True,
+        "build_testing": False,
+        "build_unbundled": False,
+    }
+
+    def configure(self):
+        self.options["libuuid"].shared = True
+        
+    def requirements(self):
+        self.requires("protobuf/3.21.12")
+        self.requires("up-cpp/0.1.2-dev")
+        self.requires("rapidjson/cci.20230929")
```

Comment:
All the build files can be built publicly

Reply:
I stand corrected :sweat_smile: 

The same convention had been used for internal custom builds of packages that otherwise used different version conventions. I'll remember to check next time